### ISSUE TITLE
Occasional crash in recent commit 40dd03e

### ISSUE THREAD

**Issue by paramat** [minetest/minetest#6390]
Happened to me 3 times but only every 10 mins or so, and when running a lua mapgen, so it's possible the lua mapgen is doing something unusual to cause this (although i can't see how).
Have not yet run under gdb or a debug build or tried to reproduce in latest master or without the mod running, but posting this in case it is useful.

Backtrace printed to terminal:
```
*** Error in `./minetest': malloc(): memory corruption (fast): 0x00007f3c4c0a14af ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7f3c6099a7e5]
/lib/x86_64-linux-gnu/libc.so.6(+0x82651)[0x7f3c609a5651]
/lib/x86_64-linux-gnu/libc.so.6(__libc_malloc+0x54)[0x7f3c609a7184]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_Znwm+0x18)[0x7f3c61299e78]
./minetest(_ZN13ScopeProfilerC1EP8ProfilerRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE17ScopeProfilerType+0xe5)[0x8f2995]
./minetest(_ZN6Server11ProcessDataEP13NetworkPacket+0x74)[0x919af4]
./minetest(_ZN6Server7ReceiveEv+0x76)[0x91e946]
./minetest(_ZN12ServerThread3runEv+0x5a)[0x92ad2a]
./minetest(_ZN6Thread10threadProcEPS_+0x49)[0x5f64d9]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0xb8c80)[0x7f3c612c4c80]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba)[0x7f3c61cb06ba]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f3c60a2a3dd]
======= Memory map: ========
00400000-00ac6000 r-xp 00000000 08:01 35914320                           /home/matstandard/minetestf/gitmods/minetest/bin/minetest
00cc5000-00cc6000 r--p 006c5000 08:01 35914320                           /home/matstandard/minetestf/gitmods/minetest/bin/minetest
00cc6000-00cc9000 rw-p 006c6000 08:01 35914320                           /home/matstandard/minetestf/gitmods/minetest/bin/minetest
00cc9000-00cf8000 rw-p 00000000 00:00 0 
014ca000-0a8db000 rw-p 00000000 00:00 0                                  [heap]
7f3c08000000-7f3c0bffd000 rw-p 00000000 00:00 0 
7f3c0bffd000-7f3c0c000000 ---p 00000000 00:00 0 
7f3c10000000-7f3c13ffe000 rw-p 00000000 00:00 0 
7f3c13ffe000-7f3c14000000 ---p 00000000 00:00 0 
7f3c14000000-7f3c16766000 rw-p 00000000 00:00 0 
7f3c16766000-7f3c18000000 ---p 00000000 00:00 0 
7f3c18000000-7f3c18021000 rw-p 00000000 00:00 0 
7f3c18021000-7f3c1c000000 ---p 00000000 00:00 0 
7f3c1c000000-7f3c20000000 rw-p 00000000 00:00 0 
7f3c24000000-7f3c266aa000 rw-p 00000000 00:00 0 
7f3c266aa000-7f3c28000000 ---p 00000000 00:00 0 
7f3c28000000-7f3c2802c000 rw-p 00000000 00:00 0 
7f3c2802c000-7f3c2c000000 ---p 00000000 00:00 0 
7f3c2e5c6000-7f3c2e5d6000 rw-s 103188000 00:06 322                       /dev/dri/card0
7f3c2e5d6000-7f3c2e5e6000 rw-s 103178000 00:06 322                       /dev/dri/card0
7f3c2e5e6000-7f3c2e5f6000 rw-s 103168000 00:06 322                       /dev/dri/card0
7f3c2e5f6000-7f3c2e606000 rw-s 103158000 00:06 322                       /dev/dri/card0
7f3c2e606000-7f3c2e616000 rw-s 103148000 00:06 322                       /dev/dri/card0
7f3c2e616000-7f3c2e626000 rw-s 103138000 00:06 322                       /dev/dri/card0
7f3c2e626000-7f3c2e636000 rw-s 103128000 00:06 322                       /dev/dri/card0
7f3c2e636000-7f3c2e646000 rw-s 103118000 00:06 322                       /dev/dri/card0
7f3c2e646000-7f3c2e656000 rw-s 103108000 00:06 322                       /dev/dri/card0
7f3c2e656000-7f3c2e666000 rw-s 1030f8000 00:06 322                       /dev/dri/card0
7f3c2e666000-7f3c2e676000 rw-s 1030e8000 00:06 322                       /dev/dri/card0
7f3c2e676000-7f3c2e686000 rw-s 1030d8000 00:06 322                       /dev/dri/card0
7f3c2e686000-7f3c2e696000 rw-s 1030c8000 00:06 322                       /dev/dri/card0
7f3c2e696000-7f3c2e6a6000 rw-s 1030b8000 00:06 322                       /dev/dri/card0
7f3c2e6a6000-7f3c2e6b6000 rw-s 1030a8000 00:06 322                       /dev/dri/card0
7f3c2e6b6000-7f3c2e6c6000 rw-s 103098000 00:06 322                       /dev/dri/card0
7f3c2e6c6000-7f3c2e6d6000 rw-s 103088000 00:06 322                       /dev/dri/card0
7f3c2e6d6000-7f3c2e6e6000 rw-s 103078000 00:06 322                       /dev/dri/card0
7f3c2e6e6000-7f3c2e6f6000 rw-s 103068000 00:06 322                       /dev/dri/card0
7f3c2e6f6000-7f3c2e706000 rw-s 103058000 00:06 322                       /dev/dri/card0
7f3c2e706000-7f3c2e716000 rw-s 103048000 00:06 322                       /dev/dri/card0
7f3c2e716000-7f3c2e726000 rw-s 103038000 00:06 322                       /dev/dri/card0
7f3c2e726000-7f3c2e736000 rw-s 103028000 00:06 322                       /dev/dri/card0
7f3c2e736000-7f3c2e746000 rw-s 103018000 00:06 322                       /dev/dri/card0
7f3c2e746000-7f3c2e756000 rw-s 103008000 00:06 322                       /dev/dri/card0
7f3c2e756000-7f3c2e766000 rw-s 102ff8000 00:06 322                       /dev/dri/card0
7f3c2e766000-7f3c2e776000 rw-s 102fe8000 00:06 322                       /dev/dri/card0
7f3c2e776000-7f3c2e786000 rw-s 102fd8000 00:06 322                       /dev/dri/card0
7f3c2e786000-7f3c2e796000 rw-s 102fc8000 00:06 322                       /dev/dri/card0
7f3c2e796000-7f3c2e7a6000 rw-s 102fb8000 00:06 322                       /dev/dri/card0
7f3c2e7a6000-7f3c2e7b6000 rw-s 102fa8000 00:06 322                       /dev/dri/card0
7f3c2e7b6000-7f3c2e7c6000 rw-s 102f98000 00:06 322                       /dev/dri/card0
7f3c2e7c6000-7f3c2e7d6000 rw-s 102f88000 00:06 322                       /dev/dri/card0
7f3c2e7d6000-7f3c2e7e6000 rw-s 102f78000 00:06 322                       /dev/dri/card0
7f3c2e7e6000-7f3c2e7f6000 rw-s 102f68000 00:06 322                       /dev/dri/card0
7f3c2e7f6000-7f3c2e806000 rw-s 00000000 00:05 494229                     /drm mm object (deleted)
7f3c2e806000-7f3c2e816000 rw-s 00000000 00:05 494228                     /drm mm object (deleted)
7f3c2e816000-7f3c2e826000 rw-s 00000000 00:05 494227                     /drm mm object (deleted)
7f3c2e826000-7f3c2e836000 rw-s 00000000 00:05 494226                     /drm mm object (deleted)
7f3c2e836000-7f3c2e846000 rw-s 00000000 00:05 494224                     /drm mm object (deleted)
7f3c2e846000-7f3c2e856000 rw-s 00000000 00:05 494223                     /drm mm object (deleted)
7f3c2e856000-7f3c2e866000 rw-s 00000000 00:05 494222                     /drm mm object (deleted)
7f3c2e866000-7f3c2e876000 rw-s 00000000 00:05 494221                     /drm mm object (deleted)
7f3c2e876000-7f3c2e886000 rw-s 00000000 00:05 494219                     /drm mm object (deleted)
7f3c2e886000-7f3c2e896000 rw-s 00000000 00:05 494218                     /drm mm object (deleted)
7f3c2e896000-7f3c2e8a6000 rw-s 00000000 00:05 494217                     /drm mm object (deleted)
7f3c2e8a6000-7f3c2e8b6000 rw-s 00000000 00:05 494216                     /drm mm object (deleted)
7f3c2e8b6000-7f3c2e8c6000 rw-s 00000000 00:05 494215                     /drm mm object (deleted)
7f3c2e8c6000-7f3c2e8d6000 rw-s 00000000 00:05 494214                     /drm mm object (deleted)
7f3c2e8d6000-7f3c2e8e6000 rw-s 00000000 00:05 494213                     /drm mm object (deleted)
7f3c2e8e6000-7f3c2e8f6000 rw-s 00000000 00:05 494212                     /drm mm object (deleted)
7f3c2e8f6000-7f3c2e906000 rw-s 00000000 00:05 494211                     /drm mm object (deleted)
7f3c2e906000-7f3c2e916000 rw-s 00000000 00:05 494210                     /drm mm object (deleted)
7f3c2e916000-7f3c2e926000 rw-s 00000000 00:05 494209                     /drm mm object (deleted)
7f3c2e926000-7f3c2e936000 rw-s 00000000 00:05 494208                     /drm mm object (deleted)
7f3c2e936000-7f3c2e946000 rw-s 102f58000 00:06 322                       /dev/dri/card0
7f3c2e946000-7f3c2e956000 rw-s 00000000 00:05 494204                     /drm mm object (deleted)
7f3c2e956000-7f3c2e966000 rw-s 00000000 00:05 494203                     /drm mm object (deleted)
7f3c2e966000-7f3c2e976000 rw-s 00000000 00:05 494202                     /drm mm object (deleted)
7f3c2e976000-7f3c2e986000 rw-s 00000000 00:05 494201                     /drm mm object (deleted)
7f3c2e986000-7f3c2e996000 rw-s 00000000 00:05 494200                     /drm mm object (deleted)
7f3c2e996000-7f3c2e9a6000 rw-s 00000000 00:05 492116                     /drm mm object (deleted)
7f3c2e9a6000-7f3c2e9b6000 rw-s 00000000 00:05 492115                     /drm mm object (deleted)
7f3c2e9b6000-7f3c2e9c6000 rw-s 00000000 00:05 492114                     /drm mm object (deleted)
7f3c2e9c6000-7f3c2e9d6000 rw-s 00000000 00:05 492113                     /drm mm object (deleted)
7f3c2e9d6000-7f3c2e9e6000 rw-s 00000000 00:05 492110                     /drm mm object (deleted)
7f3c2e9e6000-7f3c2e9f6000 rw-s 102f48000 00:06 322                       /dev/dri/card0
7f3c2e9f6000-7f3c2ea06000 rw-s 102f38000 00:06 322                       /dev/dri/card0
7f3c2ea06000-7f3c2ea16000 rw-s 102f28000 00:06 322                       /dev/dri/card0
7f3c2ea16000-7f3c2ea26000 rw-s 102f18000 00:06 322                       /dev/dri/card0
7f3c2ea26000-7f3c2ea36000 rw-s 102f08000 00:06 322                       /dev/dri/card0
7f3c2ea36000-7f3c2ea46000 rw-s 102ef8000 00:06 322                       /dev/dri/card0
7f3c2ea46000-7f3c2ea56000 rw-s 102ee8000 00:06 322                       /dev/dri/card0
7f3c2ea56000-7f3c2ea66000 rw-s 102ed8000 00:06 322                       /dev/dri/card0
7f3c2ea66000-7f3c2ea6e000 rw-s 00000000 00:05 494826                     /drm mm object (deleted)
7f3c2ea6e000-7f3c2ea7e000 rw-s 102ec8000 00:06 322                       /dev/dri/card0
7f3c2ea7e000-7f3c2ea86000 rw-s 00000000 00:05 493379                     /drm mm object (deleted)
7f3c2ea86000-7f3c2ea96000 rw-s 102eb8000 00:06 322                       /dev/dri/card0
7f3c2ea96000-7f3c2eaa6000 rw-s 102ea8000 00:06 322                       /dev/dri/card0
7f3c2eaa6000-7f3c2eab6000 rw-s 102e98000 00:06 322                       /dev/dri/card0
7f3c2eab6000-7f3c2eac6000 rw-s 102e88000 00:06 322                       /dev/dri/card0
7f3c2eac6000-7f3c2ead6000 rw-s 102e78000 00:06 322                       /dev/dri/card0
7f3c2ead6000-7f3c2eae6000 rw-s 00000000 00:05 492002                     /drm mm object (deleted)
7f3c2eae6000-7f3c2eaf6000 rw-s 00000000 00:05 492001                     /drm mm object (deleted)
7f3c2eaf6000-7f3c2eb06000 rw-s 00000000 00:05 492000                     /drm mm object (deleted)
7f3c2eb06000-7f3c2eb16000 rw-s 00000000 00:05 491997                     /drm mm object (deleted)
7f3c2eb16000-7f3c2eb26000 rw-s 00000000 00:05 491995                     /drm mm object (deleted)
7f3c2eb26000-7f3c2eb36000 rw-s 00000000 00:05 494825                     /drm mm object (deleted)
7f3c2eb36000-7f3c2eb46000 rw-s 00000000 00:05 494180                     /drm mm object (deleted)
7f3c2eb46000-7f3c2eb56000 rw-s 00000000 00:05 494179                     /drm mm object (deleted)
7f3c2eb56000-7f3c2eb66000 rw-s 00000000 00:05 493377                     /drm mm object (deleted)
7f3c2eb66000-7f3c2eb76000 rw-s 00000000 00:05 493372                     /drm mm object (deleted)
7f3c2eb76000-7f3c2eb86000 rw-s 00000000 00:05 493371                     /drm mm object (deleted)
7f3c2eb86000-7f3c2eb96000 rw-s 00000000 00:05 493370                     /drm mm object (deleted)
7f3c2eb96000-7f3c2eba6000 rw-s 00000000 00:05 493369                     /drm mm object (deleted)
7f3c2eba6000-7f3c2ebb6000 rw-s 00000000 00:05 493354                     /drm mm object (deleted)
7f3c2ebb6000-7f3c2ebc6000 rw-s 00000000 00:05 493353                     /drm mm object (deleted)
7f3c2ebc6000-7f3c2ebd6000 rw-s 00000000 00:05 493352                     /drm mm object (deleted)
7f3c2ebd6000-7f3c2ebe6000 rw-s 00000000 00:05 493349                     /drm mm object (deleted)
7f3c2ebe6000-7f3c2ebf6000 rw-s 00000000 00:05 493346                     /drm mm object (deleted)
7f3c2ebf6000-7f3c2ec06000 rw-s 102e68000 00:06 322                       /dev/dri/card0
7f3c2ec06000-7f3c2ec16000 rw-s 102e58000 00:06 322                       /dev/dri/card0
7f3c2ec16000-7f3c2ec26000 rw-s 102e48000 00:06 322                       /dev/dri/card0
7f3c2ec26000-7f3c2ec36000 rw-s 102e38000 00:06 322                       /dev/dri/card0
7f3c2ec36000-7f3c2ec3e000 rw-s 00000000 00:05 494176                     /drm mm object (deleted)
7f3c2ec3e000-7f3c2ec4e000 rw-s 00000000 00:05 493304                     /drm mm object (deleted)
7f3c2ec4e000-7f3c2ec5e000 rw-s 102e28000 00:06 322                       /dev/dri/card0
7f3c2ec5e000-7f3c2ec6e000 rw-s 102e18000 00:06 322                       /dev/dri/card0
7f3c2ec6e000-7f3c2ec7e000 rw-s 00000000 00:05 491992                     /drm mm object (deleted)
7f3c2ec7e000-7f3c2ec8e000 rw-s 102e08000 00:06 322                       /dev/dri/card0
7f3c2ec8e000-7f3c2ec9e000 rw-s 102df8000 00:06 322                       /dev/dri/card0
7f3c2ec9e000-7f3c2eca6000 rw-s 00000000 00:05 494817                     /drm mm object (deleted)
7f3c2eca6000-7f3c2ecb6000 rw-s 00000000 00:05 491991                     /drm mm object (deleted)
7f3c2ecb6000-7f3c2ecc6000 rw-s 00000000 00:05 494174                     /drm mm object (deleted)
7f3c2ecc6000-7f3c2ecd6000 rw-s 00000000 00:05 494173                     /drm mm object (deleted)
7f3c2ecd6000-7f3c2ece6000 rw-s 00000000 00:05 494172                     /drm mm object (deleted)
7f3c2ece6000-7f3c2ecf6000 rw-s 00000000 00:05 494170                     /drm mm object (deleted)
7f3c2ecf6000-7f3c2ed06000 rw-s 00000000 00:05 494169                     /drm mm object (deleted)
7f3c2ed06000-7f3c2ed0e000 rw-s 00000000 00:05 494162                     /drm mm object (deleted)
7f3c2ed0e000-7f3c2ed1e000 rw-s 00000000 00:05 491990                     /drm mm object (deleted)
7f3c2ed1e000-7f3c2ed2e000 rw-s 102de8000 00:06 322                       /dev/dri/card0
7f3c2ed2e000-7f3c2ed3e000 rw-s 00000000 00:05 493297                     /drm mm object (deleted)
7f3c2ed3e000-7f3c2ed4e000 rw-s 102dd8000 00:06 322                       /dev/dri/card0
7f3c2ed4e000-7f3c2ed5e000 rw-s 102dc8000 00:06 322                       /dev/dri/card0
7f3c2ed5e000-7f3c2ed6e000 rw-s 00000000 00:05 494160                     /drm mm object (deleted)
7f3c2ed6e000-7f3c2ed7e000 rw-s 102db8000 00:06 322                       /dev/dri/card0
7f3c2ed7e000-7f3c2ed8e000 rw-s 00000000 00:05 493291                     /drm mm object (deleted)
7f3c2ed8e000-7f3c2ed9e000 rw-s 00000000 00:05 491519                     /drm mm object (deleted)
7f3c2ed9e000-7f3c2edae000 rw-s 00000000 00:05 494150                     /drm mm object (deleted)
7f3c2edae000-7f3c2edbe000 rw-s 102da8000 00:06 322                       /dev/dri/card0
7f3c2edbe000-7f3c2edce000 rw-s 102d98000 00:06 322                       /dev/dri/card0
7f3c2edce000-7f3c2edde000 rw-s 00000000 00:05 491517                     /drm mm object (deleted)
7f3c2edde000-7f3c2edee000 rw-s 102d88000 00:06 322                       /dev/dri/card0
Aborted (core dumped)
```

**Comment by paramat** [comment]
Probably fixed by a3c298e1d1f02634e87c7ba1746fd7e9ed444875 will close soon.

**Comment by paramat** [comment]
Updated to latest master and no crashes so far, probably fixed by a recent commit as nerzhul informed me.
